common/rpc/factory.go

```diff
@@ -37,16 +37,15 @@ const defaultGRPCSizeLimit = 4 * 1024 * 1024
 
 // Factory is an implementation of common.RPCFactory interface
 type Factory struct {
-	maxMessageSize    int
-	hostAddressMapper HostAddressMapper
-	channel           tchannel.Channel
-	dispatcher        *yarpc.Dispatcher
+	maxMessageSize int
+	channel        tchannel.Channel
+	dispatcher     *yarpc.Dispatcher
 }
 
 // NewFactory builds a new rpc.Factory
 func NewFactory(logger log.Logger, p Params) *Factory {
 	inbounds := yarpc.Inbounds{}
-
+	logger = logger.WithTags(tag.Service(p.ServiceName))
```

Comment:
I believe we tag service name here already: https://github.com/uber/cadence/blob/7084679d54d983c7667d319ac3321c97d61d3e3f/common/resource/params.go#L74

Although I don't particularly like such tagging within Update... function.

Reply:
removed `UpdateLoggerWithServiceName` and logger tagging in rpc factory